lib/Analysis/LiveValueAnalysis.cpp

```diff
@@ -0,0 +1,1321 @@
+
+
+#include "llvm/Analysis/LiveValueAnalysis.h"
+#include "llvm/Analysis/Passes.h"
+#include "llvm/IR/CFG.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/Dominators.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/IntrinsicInst.h"
+#include "dxc/DXIL/DxilMetadataHelper.h"
+#include "dxc/Support/Global.h"
+
+#include <fstream>
+
+using namespace llvm;
+
+// Register this pass...
+char LiveValueAnalysis::ID = 0;
+INITIALIZE_PASS(LiveValueAnalysis, "lva", "Live Value Analysis and reporting for DXR", false, true)
+
+ModulePass *llvm::createLiveValueAnalysisPass(StringRef LiveValueAnalysisOutputFile) {
+  return new LiveValueAnalysis(LiveValueAnalysisOutputFile);
+}
+
+LiveValueAnalysis::LiveValueAnalysis(StringRef LiveValueAnalysisOutputFile = "")
```

Comment:
The default value needs to be in the method prototype. Linux compilers won't tolerate it here.

Reply:
Thanks, fixed.